rskj-core/src/test/java/co/rsk/peg/bitcoin/P2shP2wshErpCustomRedeemScriptBuilderTest.java

```diff
@@ -0,0 +1,208 @@
+        );
+    }
+
+    private static Stream<Arguments> invalidInputsArgsProvider() {
+        long surpassingMaxCsvValue = ErpRedeemScriptBuilderUtils.MAX_CSV_VALUE + 1;
+        return Stream.of(
+            // defaultKeys is null
+            Arguments.of(NullPointerException.class, null, 0, emergencyKeys, ERP_THRESHOLD, CSV_VALUE),
+            // defaultThreshold is zero
+            Arguments.of(IllegalArgumentException.class, defaultKeys, 0, emergencyKeys, ERP_THRESHOLD, CSV_VALUE),
+            // defaultThreshold is negative
+            Arguments.of(IllegalArgumentException.class, defaultKeys, -1, emergencyKeys, ERP_THRESHOLD, CSV_VALUE),
+            // empty default keys
+            Arguments.of(IllegalArgumentException.class, Collections.emptyList(), 0, emergencyKeys, ERP_THRESHOLD, CSV_VALUE),
+            // threshold greater than default keys size
+            Arguments.of(IllegalArgumentException.class, defaultKeys, defaultKeys.size() + 1, emergencyKeys, ERP_THRESHOLD, CSV_VALUE),
+            // erpKeys is null
+            Arguments.of(NullPointerException.class, defaultKeys, DEFAULT_THRESHOLD, null, ERP_THRESHOLD, CSV_VALUE),
+            // empty erp keys
+            Arguments.of(IllegalArgumentException.class, defaultKeys, DEFAULT_THRESHOLD, Collections.emptyList(), ERP_THRESHOLD, CSV_VALUE),
+            // erp threshold is negative
+            Arguments.of(IllegalArgumentException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, -1, CSV_VALUE),
+            // erp threshold greater than erp keys size
+            Arguments.of(IllegalArgumentException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, emergencyKeys.size() + 1, CSV_VALUE),
+            // csv is negative
+            Arguments.of(RedeemScriptCreationException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, ERP_THRESHOLD, -1L),
+            // csv is zero
+            Arguments.of(RedeemScriptCreationException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, ERP_THRESHOLD, 0L),
+            // csv is above the maximum allowed
+            Arguments.of(RedeemScriptCreationException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, ERP_THRESHOLD, surpassingMaxCsvValue)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("providePubKeysAndThresholds")
+    void of_shouldHaveTheCorrectRedeemScript(List<BtcECKey> defaultKeys, int threshold) {
+        // Act
+        Script redeemScript = builder.of(
+            defaultKeys, threshold, emergencyKeys, ERP_THRESHOLD, CSV_VALUE
+        );
+
+        // Assert
+        byte[] p2shp2wshErpCustomRedeemScriptProgram = redeemScript.getProgram();
+
+        // redeemScript - First opcode should be the OP_NOTIF
+        int opNotIfIndex = 0;
+        byte actualOpCode = p2shp2wshErpCustomRedeemScriptProgram[opNotIfIndex];
+
+        assertEquals((byte) ScriptOpCodes.OP_NOTIF, actualOpCode);
+
+        // defaultCustomRedeemScript - Second byte should be the PubKey
```

Comment:
```suggestion
        // defaultCustomRedeemScript
```

repeated below right?

Reply:
yes, but still the second byte is the pub key

Reply:
lol you right